pkg/datamover/profile.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kopia/snapshot"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+type Profile struct {
+	outputName string
+	profile    *param.Profile
+	snapJSON   string
+}
+
+func (p *Profile) Pull(ctx context.Context, sourcePath, destinationPath string) error {
+	if p.profile.Location.Type == crv1alpha1.LocationTypeKopia {
+		if p.snapJSON == "" {
+			return errors.New("kopia snapshot information is required to pull data using kopia")
+		}
+		kopiaSnap, err := snapshot.UnmarshalKopiaSnapshot(p.snapJSON)
+		if err != nil {
+			return err
+		}
+		if err = p.connectToKopiaRepositoryServer(ctx); err != nil {
+			return err
+		}
```

Comment:
This snippet is repeated in Delete() also. We should consider moving this common code to a helper function

Reply:
Done !